File: core/logger.py

import sys
from config import settings

class Logger:
    def header(self, message):
        print(f"\n{settings.Colors.HEADER}{settings.Colors.BOLD}=== {message} ==={settings.Colors.ENDC}")

    def info(self, message):
        print(f"{settings.Colors.OKBLUE}[INFO]{settings.Colors.ENDC} {message}")

    def success(self, message):
        print(f"{settings.Colors.OKGREEN}[OK]{settings.Colors.ENDC} {message}")

    def warning(self, message):
        print(f"{settings.Colors.WARNING}[WARN]{settings.Colors.ENDC} {message}")

    def fail(self, message):
        print(f"{settings.Colors.FAIL}[FAIL]{settings.Colors.ENDC} {message}")

logger = Logger()

File: Core/Logger.cs
using Spectre.Console;

namespace UniSentinel.Core;

internal static class Logger
{
    private static readonly Random t_rnd = new();
    private static readonly char[] t_noise = ['!', '@', '#', '$', '%', '^', '&', '*', '?', 'X', 'Z', '0', '1', '░', '▒', '▓'];

    public static void Header(string m)
    {
        var rule = new Rule($"[bold yellow]{m.ToUpperInvariant()}[/]")
        {
            Justification = Justify.Left
        };
        AnsiConsole.Write(rule);
    }

    public static void Success(string m) => AnsiConsole.MarkupLine($"[bold green][[OK]][/] {Markup.Escape(m)}");
    public static void Fail(string m) => AnsiConsole.MarkupLine($"[bold red][[ERR]][/] {Markup.Escape(m)}");
    public static void Warning(string m) => AnsiConsole.MarkupLine($"[bold yellow][[!]][/] {Markup.Escape(m)}");

    public static void Info(string m)
    {
        AnsiConsole.Markup($"[grey][[...]][/] ");
        TypeEffect(m);
        Console.WriteLine();
    }

    private static void TypeEffect(string text)
    {
        foreach (var c in text)
        {
            Console.Write(t_noise[t_rnd.Next(t_noise.Length)]);
            Thread.Sleep(3);
            Console.Write("\b");
            Console.Write(c);
            Thread.Sleep(2);
        }
    }
}
